crucible-wasm/test/Test.hs

```diff
@@ -0,0 +1,153 @@
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Main where
+
+import           Control.Exception ( SomeException, catches, try, Handler(..), IOException )
+import           Control.Lens ( (^.), (^?), _Right, to )
+import           Control.Monad ( unless, when )
+import           Data.Bifunctor ( first )
+import qualified Data.ByteString.Lazy as BSIO
+import qualified Data.ByteString.Lazy.Char8 as BSC
+import           Data.Char ( isLetter, isSpace )
+import           Data.List.Extra ( isInfixOf, isPrefixOf, stripPrefix )
+import           Data.Maybe ( catMaybes, fromMaybe, isJust )
+import qualified Data.Text as T
+import           Data.Versions ( Versioning, versioning, prettyV, major )
+import qualified GHC.IO.Exception as GE
+import           Numeric.Natural
+import           System.Environment ( withArgs, lookupEnv )
+import           System.Exit (ExitCode(..))
+import           System.FilePath ( (-<.>) )
+import qualified System.IO as IO
+
+import qualified Test.Tasty as TT
+import           Test.Tasty.HUnit ( testCase, assertFailure )
+import qualified Test.Tasty.Sugar as TS
+
+import qualified Crux
+
+import qualified Lang.Crucible.Wasm.Main as C
+
+
+cube :: TS.CUBE
+cube =
+  TS.mkCUBE
+  { TS.inputDirs = ["test-data/"]
+  , TS.rootName = "*.txt"
```

Comment:
If I'm reading [this part of the WebAssembly spec](https://webassembly.github.io/spec/core/text/conventions.html) correctly, the recommended file extension for WebAssembly code is `.wat`. We should probably use that here as well.